pkg/kubectl/cmd/rollingupdate.go

```diff
@@ -120,10 +121,13 @@ func RunRollingUpdate(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, arg
 	}
 
 	var hasLabel bool
-	for key, oldValue := range oldRc.Spec.Selector {
-		if newValue, ok := newRc.Spec.Selector[key]; ok && newValue != oldValue {
-			hasLabel = true
-			break
+	for _, oldReq := range oldRc.Spec.Selector {
+		for _, newReq := range newRc.Spec.Selector {
+			if oldReq.Key == newReq.Key &&
+				!reflect.DeepEqual(oldReq.StrValues, newReq.StrValues) {
+				hasLabel = true // values are different for the same key
+				break
```

Comment:
this break only breaks the inner loop.  Seems you want to break the outer loop too..